File: dto/auth_dto.go
package dto

import (
	"github.com/google/uuid"
)

type RegisterRequest struct {
	FirstName string `json:"first_name" binding:"required"`
	LastName  string `json:"last_name" binding:"required"`
	Gender    string `json:"gender"`
	DOB       string `json:"dob"` // YYYY-MM-DD
	Mobile    string `json:"mobile" binding:"required"`
	Email     string `json:"email" binding:"required,email"`
	Password  string `json:"password" binding:"required"`
}

type RegisterResponse struct {
	Message          string    `json:"message"`
	PublicID         uuid.UUID `json:"public_id"`
	Mobile           string    `json:"mobile"`
	ExpiresInSeconds int       `json:"expires_in_seconds"`
}

type VerifyOTPRequest struct {
	PublicID string `json:"public_id"`
	Mobile   string `json:"mobile"`
	OTP      string `json:"otp" binding:"required"`
	Purpose  string `json:"purpose" binding:"required"`
}

type LoginRequest struct {
	Identifier string `json:"identifier" binding:"required"` // Mobile or Email
	Password   string `json:"password" binding:"required"`
}

type DoctorProfileDTO struct {
	PublicID       uuid.UUID `json:"public_id"`
	FirstName      string    `json:"first_name"`
	LastName       string    `json:"last_name"`
	Email          string    `json:"email"`
	Mobile         string    `json:"mobile"`
	Status         string    `json:"status"`
	MobileVerified bool      `json:"mobile_verified"`
	EmailVerified  bool      `json:"email_verified"`
}

type AuthResponse struct {
	Message        string            `json:"message"`
	AccessToken    string            `json:"access_token,omitempty"`
	RefreshToken   string            `json:"refresh_token,omitempty"`
	TokenType      string            `json:"token_type,omitempty"`
	ExpiresIn      int64             `json:"expires_in,omitempty"`
	MobileVerified bool              `json:"mobile_verified"`
	Doctor         *DoctorProfileDTO `json:"doctor,omitempty"`
}

File: python_backend/dto/auth_dto.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID

class RegisterRequest(BaseModel):
    first_name: str = Field(...)
    last_name: str = Field(...)
    gender: Optional[str] = None
    dob: Optional[str] = Field(None, description="YYYY-MM-DD")
    mobile: str = Field(...)
    email: str = Field(...)
    password: str = Field(...)

class RegisterResponse(BaseModel):
    message: str
    public_id: UUID
    mobile: str
    expires_in_seconds: int

class VerifyOTPRequest(BaseModel):
    public_id: Optional[str] = None
    mobile: Optional[str] = None
    otp: str = Field(...)
    purpose: str = Field(...)

class LoginRequest(BaseModel):
    identifier: str = Field(...)
    password: str = Field(...)

class DoctorProfileDTO(BaseModel):
    public_id: UUID
    first_name: str
    last_name: str
    email: str
    mobile: str
    status: str
    mobile_verified: bool
    email_verified: bool

class AuthResponse(BaseModel):
    message: str
    access_token: Optional[str] = None
    refresh_token: Optional[str] = None
    token_type: Optional[str] = None
    expires_in: Optional[int] = None
    mobile_verified: bool
    doctor: Optional[DoctorProfileDTO] = None
